### ISSUE TITLE
Cannot resolve intranet DNS names inside PODs

### ISSUE THREAD

**Issue by maslick** [openshift/origin#22144]
I have deployed an openshift cluster on one of the machines (Ubuntu 18.04) inside a corporate network by running:
```
$ oc cluster up --public-hostname=dev3.maslick.com --routing-suffix=dev3.maslick.com
```

All works fine, however if I want to create a new app from a private repo (hosted internally), say:
```
$ oc new-app https://gitlab.maslick.com/test/hello-world.git
```

I get:
```
Cloning "https://gitlab.maslick.com/test/hello-world.git " ...
--
  | error: fatal: unable to access 'https://gitlab.maslick.com/test/hello-world.git/':  Could not resolve host: gitlab.maslick.com; Unknown error
```

I can resolve internal DNS names from the ``dev3`` machine, also from Docker containers on ``dev3``. I suspect I have to change some DNS related setting in one of the yaml files inside the ``openshift.local.clusterup`` directory created by ``oc cluster up``. Any ideas? Thanks.
___
##### Version
```
~ oc version
oc v3.11.0+0cbc58b
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://dev3.maslick.com:8443
kubernetes v1.11.0+d4cacc0
```

##### Steps To Reproduce
see at the top

##### Current Result
Unable to resolve dns names (i.e. private gitlab, etc) from PODs.
##### Expected Result
The ability to resolve internal dns names from PODs:
```
$ cat /etc/resolv.conf
nameserver 172.30.0.2
nameserver 4.3.2.1         # <- sth like this where 4.3.2.1 is my local DNS server
search test-project.svc.cluster.local svc.cluster.local cluster.local maslick.com
options ndots:5
```

##### Additional Information
```
$ oc adm diagnostics
ERROR: [DCli2012 from diagnostic DiagnosticPod@openshift/origin/pkg/oc/cli/admin/diagnostics/diagnostics/client/pod/run_diagnostics_pod.go:208]
       See the errors below in the output from the diagnostic pod:
       [Note] Running diagnostic: PodCheckAuth
              Description: Check that service account credentials authenticate as expected

       Info:  Service account token successfully authenticated to master

       ERROR: [DP1014 from diagnostic PodCheckAuth@openshift/origin/pkg/oc/cli/admin/diagnostics/diagnostics/client/pod/in_pod/auth.go:172]
              Request to integrated registry timed out; this typically indicates network or SDN problems.

       [Note] Running diagnostic: PodCheckDns
              Description: Check that DNS within a pod works as expected

       [Note] Summary of diagnostics execution (version v3.11.0+d0c29df-98):
       [Note] Errors seen: 1
```
``resolv.conf`` on ``dev3``:
```
$ cat /etc/resolv.conf
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
# 127.0.0.53 is the systemd-resolved stub resolver.
# run "systemd-resolve --status" to see details about the actual nameservers.

search maslick.com
nameserver 4.3.2.1             # <-- corporate DNS server
```

``resolv.conf`` inside Docker container:
```
$ docker run -it --rm  busybox
/ # cat /etc/resolv.conf
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
# 127.0.0.53 is the systemd-resolved stub resolver.
# run "systemd-resolve --status" to see details about the actual nameservers.

search maslick.com
nameserver 4.3.2.1
```

``resolv.conf`` inside an Openshift POD:
```
$ cat /etc/resolv.conf
nameserver 172.30.0.2
search test-project.svc.cluster.local svc.cluster.local cluster.local maslick.com
options ndots:5
```

**Comment by maslick** [comment]
Seems, the issue is gone once I run ``oc cluster up --public-hostname=dev3.maslick.com --routing-suffix=dev3.maslick.com`` from another directory.

Apparently, the first cluster used the old settings of the master host (I edited ``/etc/resolv.conf``), and didn't pick up the new ones.

The POD's resolv.conf looks like this now (no changes):
```
$ cat /etc/resolv.conf
nameserver 172.30.0.2
search myproject.svc.cluster.local svc.cluster.local cluster.local maslick.com
options ndots:5
```
But it resolves internal dns names nonetheless.

**Comment by maslick** [comment]
Just out of curiosity, how do internal dns names get resolved now if there's only one nameserver (172.30.0.2) in the PODs' ``resolv.conf``? Technically.
